### PR TITLE
fix: last plugin release version and opencart version

### PR DESCRIPTION
**Pull request by DiegoHBustamante**
This PR add the last plugin release version and Opencart version

> Before the changes
Captura de pantalla 2024-07-30 a la(s) 1 07 43 p  m

> After the changes
Captura de pantalla 2024-07-30 a la(s) 8 05 47 p  m

### CODE REVIEW

**Comment by mastudillot** [comment]
Fix the sonar issues

File: src/upload/catalog/controller/extension/payment/libwebpay_rest/HealthCheck.php [hunk 1]
```diff
@@ -91,15 +91,29 @@ private function getLastGitHubReleaseVersion($string){
...
+        curl_setopt($ch,CURLOPT_URL,$baseurl);
+        curl_setopt($ch,CURLOPT_RETURNTRANSFER,true);
+        curl_setopt($ch, CURLOPT_USERAGENT, $agent);
+        $content=curl_exec($ch);
+        curl_close($ch);
+        $con = json_decode($content, true);
+        $version = array_key_exists('tag_name',$con) ? $con['tag_name'] : '';
+        return $version;
+    }
+
```
Comment (mastudillot) [inline src/upload/catalog/controller/extension/payment/libwebpay_rest/HealthCheck.php]: This function is duplicated, reuse getLastGitHubReleaseVersion

File: src/upload/catalog/controller/extension/payment/libwebpay_rest/HealthCheck.php [hunk 2]
```diff
@@ -91,15 +91,29 @@ private function getLastGitHubReleaseVersion($string){
...
+        $actualversion = $this->getLastGitHubReleaseVersion('opencart/opencart');
+        $lastversionplugin = $this->getLastGitHubReleaseVersionPlugin('TransbankDevelopers/transbank-plugin-opencart-webpay-rest');
         $currentplugin = TransbankSdkWebpay::PLUGIN_VERSION;
         $result = array(
             'current_ecommerce_version' => $actualversion,
-            'last_ecommerce_version' => $lastversion,
-            'current_plugin_version' => $currentplugin
+            'current_plugin_version' => $currentplugin,
+            'last_version_plugin' => $lastversionplugin
         );
```
Comment (mastudillot) [inline src/upload/catalog/controller/extension/payment/libwebpay_rest/HealthCheck.php]: make variable names more descriptive

**Review by mastudillot — COMMENTED** [review]
check my comments

File: src/upload/catalog/controller/extension/payment/libwebpay_rest/HealthCheck.php [hunk 3]
```diff
@@ -47,7 +49,18 @@ private function getValidatephp(){
...
+    private function getOpenCartVersion() {
+        if (file_exists($this->configFile)) {
+            $fileContent = file_get_contents($this->configFile);
+            if (preg_match("/define\('VERSION', '([^']+)'\);/", $fileContent, $matches)) {
+                return $matches[1];
+            } else {
+                throw new Exception('No se pudo encontrar la versión de OpenCart.');
+            }
+        } else {
+            throw new Exception('El archivo admin/index.php no existe.');
```
Comment (Matiasnickolas) [inline src/upload/catalog/controller/extension/payment/libwebpay_rest/HealthCheck.php]: are this exceptions controlled when this method is used? maybe can return an empty or 'not found' string instead exceptions

**Pull request closed without merging**